sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestBase.java

```diff
@@ -0,0 +1,243 @@
+            addTextReplacementRules(textReplacementRules);
+        }
+        initializeClients(httpPipeline, testProfile);
+    }
+
+    private HttpClient generateHttpClientWithProxy(ProxyOptions proxyOptions) {
+        NettyAsyncHttpClientBuilder clientBuilder = new NettyAsyncHttpClientBuilder();
+        if (proxyOptions != null) {
+            clientBuilder.proxy(proxyOptions);
+        } else {
+            try {
+                System.setProperty(USE_SYSTEM_PROXY, VALUE_TRUE);
+                List<Proxy> proxies = ProxySelector.getDefault().select(new URI(AzureEnvironment.AZURE.getResourceManagerEndpoint()));
+                if (!proxies.isEmpty()) {
+                    for (Proxy proxy : proxies) {
+                        if (proxy.address() instanceof InetSocketAddress) {
+                            String host = ((InetSocketAddress) proxy.address()).getHostName();
+                            int port = ((InetSocketAddress) proxy.address()).getPort();
+                            switch (proxy.type()) {
+                                case HTTP:
+                                    return clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress(host, port))).build();
+                                case SOCKS:
+                                    return clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.SOCKS5, new InetSocketAddress(host, port))).build();
+                                default:
+                            }
+                        }
+                    }
+                }
+                String host = null;
+                int port = 0;
+                if (System.getProperty(HTTPS_PROXY_HOST) != null && System.getProperty(HTTPS_PROXY_PORT) != null) {
+                    host = System.getProperty(HTTPS_PROXY_HOST);
+                    port = Integer.parseInt(System.getProperty(HTTPS_PROXY_PORT));
+                } else if (System.getProperty(HTTP_PROXY_HOST) != null && System.getProperty(HTTP_PROXY_PORT) != null) {
+                    host = System.getProperty(HTTP_PROXY_HOST);
+                    port = Integer.parseInt(System.getProperty(HTTP_PROXY_PORT));
+                }
+                if (host != null) {
+                    clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress(host, port)));
+                }
+            } catch (URISyntaxException e) { }
+        }
+        return clientBuilder.build();
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!isSkipInPlayback) {
+            cleanUpResources();
+        }
+    }
+
+    private void addTextReplacementRules(Map<String, String> rules) {
+        for (Map.Entry<String, String> entry : rules.entrySet()) {
+            interceptorManager.addTextReplacementRule(entry.getKey(), entry.getValue());
+        }
+    }
+
+    protected abstract HttpPipeline buildHttpPipeline(
```

Comment:
Make buildHttpPipeline as abstract as some policies like AuthenticationPolicy we have to put in resources as the dependency on resources.